### PR TITLE
Port Strategy to Huff

### PR DESCRIPTION
**Pull request by jtriley-eth**
Closes issue #6 

This ports the strategy to huff and maintains feature parity to make it a drop-in replacement. Further details are written in the README file.

### CODE REVIEW

**Comment by jtriley-eth** [comment]
Also, the contract is precompiled. A forge script can be run with ffi enabled to validate the compiled code matches that of the huff compiler. This is the only script that requires `huffc`.

File: crates/strategies/opensea-sudo-arb/contracts/README.md [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+    . bop_end_time
+    . bop_zone_hash
+    . bop_salt
+    . bop_offerer_conduit_key
+    . bop_fulfuller_conduit_key
+    . bop_total_original_additional_recipients
+    . bop_additional_recipients_ptr
+    . bop_signatures_ptr
+    . bop_additional_recipients_len
+    . (bop_additional_recipients_amount . bop_additional_recipients_recipient)+
```
Comment (FrankieIsLost) [inline crates/strategies/opensea-sudo-arb/contracts/README.md:112]: nit: is the formatting on this line intentional? was a bit confused about how to interpret, the point is that you can have more than one pair of `bop_additional_recipients_amount and bop_additional_recipients_recipient` right?

File: crates/strategies/opensea-sudo-arb/contracts/README.md [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+contract SudoOpenseaArbFast is Owned {
+
+    error NoProfit();
+
+    constructor() Owned(msg.sender) {}
+
+    Seaport constant seaport = Seaport(0x00000000000001ad428e4906aE43D8F9852d0dD6);
+
+    /// @custom:unsafe-as-fuck
+    function executeArb(BasicOrderParameters calldata, uint256 paymentValue, address payable sudo_pool) public {
```
Comment (FrankieIsLost) [inline crates/strategies/opensea-sudo-arb/contracts/README.md:202]: could we please add comments explaining what some of the constants are in the reference implementation? would make it a bit easier to follow along.

File: crates/strategies/opensea-sudo-arb/contracts/src/huff/LibCall.huff [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+/// ### Panics
+///
+/// - When `CALL_AND_BUBBLE` panics.
+#define macro CALL_SWAP() = takes (0) returns (0) {
+    [SWAP_SEL]                  // [swap_selector]
+    [SEL_MEM_PTR]               // [mem_ptr, swap_selector]
+    mstore                      // []
+
+    0xa0                        // [nftids_ptr]
+    [ARG0_MEM_PTR]              // [mem_ptr, nftids_ptr
```
Comment (FrankieIsLost) [inline crates/strategies/opensea-sudo-arb/contracts/src/huff/LibCall.huff:111]: nit: missing bracket

**Review by FrankieIsLost — COMMENTED** [review]
LFG -- this looks great, left a few smol nits

File: crates/strategies/opensea-sudo-arb/contracts/README.md (hunk 1 above)
Comment (jtriley-eth) [reply crates/strategies/opensea-sudo-arb/contracts/README.md:112]: right, `+` means one or more. would it be better to annotate this or is there a better way to describe it?

File: crates/strategies/opensea-sudo-arb/contracts/README.md (hunk 2 above)
Comment (jtriley-eth) [reply crates/strategies/opensea-sudo-arb/contracts/README.md:235]: added constant names to clarify each variables' role. lmk if there should be more context

**Review by FrankieIsLost — APPROVED** [review]
LGTM 🚀

**Pull request merged**